src/Microsoft.OData.Core/JsonLight/ODataJsonLightWriter.cs

```diff
@@ -1018,32 +1021,51 @@ protected override void WriteEntityReferenceInNavigationLinkContent(ODataNestedR
             // In JSON Light, we can only write entity reference links at the beginning of a navigation link in requests;
             // once we wrote a resource set, entity reference links are not allowed anymore (we require all the entity reference
             // link to come first because of the grouping in the JSON Light wire format).
-            JsonLightNestedResourceInfoScope navigationLinkScope = (JsonLightNestedResourceInfoScope)this.CurrentScope;
-            if (navigationLinkScope.ResourceSetWritten)
+            JsonLightNestedResourceInfoScope nestedResourceScope = this.CurrentScope as JsonLightNestedResourceInfoScope;
+            if (nestedResourceScope == null)
+            {
+                nestedResourceScope = this.ParentNestedResourceInfoScope as JsonLightNestedResourceInfoScope;

```

Comment:
>ParentNestedResourceInfoScope [](start = 43, length = 29)

This is not clear to me; if a parentNestedResourceInfo is passed, why are we using the private member? Should the names be different then?

Reply:
@KanishManuja-MS What do you mean? 

ParentNestedResourceInfoScope is used if there's a ResourceSet.